### PR TITLE
manim: fix build failure due to failing tests

### PR DESCRIPTION
**Pull request by mofrim**
###### Description of changes

had to add 2 additional tests to `failing_tests.nix`. will try to figure out where this type of failure precisely originates from.. but it might be rooted in the design of the tests itself, so it can be ignored to some extent (..there are still 544 passing tests). Any help with figuring out why the ignored tests fail would be highly appreciated.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/manim/failing_tests.nix [hunk 1]
```diff
@@ -65,6 +65,8 @@
 # E           Max relative difference: 255.
 [
   "test_Text2Color"
+  "test_PointCloudDot"
+  "test_Torus"
```
Comment (SuperSandro2000) [inline pkgs/applications/video/manim/failing_tests.nix:69]: Just to be sure: are they failing for the same reason?

**Comment by smancill** [comment]
I don't have a Linux box, and the package is marked as broken on Darwin, so I cannot try.

But I am confused, it is green on Hydra:

https://hydra.nixos.org/job/nixpkgs/trunk/manim.x86_64-linux

File: pkgs/applications/video/manim/failing_tests.nix (hunk 1 above)
Comment (friedelino) [reply pkgs/applications/video/manim/failing_tests.nix:69]: The msgs of the failing tests are not exactly equal but very similar to the above. That's why I assume there is a common cause for the failures.

1) `test_PointCloudDot` fails with:
```
E           AssertionError:
E           Not equal to tolerance rtol=1e-07, atol=1.01
E           Frame no -1. You can use --show_diff to visually show the difference.
E           Mismatched elements: 8 / 1639680 (0.000488%)
E           Max absolute difference: 255
E           Max relative difference: 0.00787402
```

2) `test_Torus` fails with:
```
E           AssertionError:
E           Not equal to tolerance rtol=1e-07, atol=1.01
E           Frame no -1. You can use --show_diff to visually show the difference.
E           Mismatched elements: 1 / 1639680 (6.1e-05%)
E           Max absolute difference: 255
E           Max relative difference: 2.42857143
```

**Comment by friedelino** [comment]
> I don't have a Linux box, and the package is marked as broken on Darwin, so I cannot try.
> 
> But I am confused, it is green on Hydra:
> 
> https://hydra.nixos.org/job/nixpkgs/trunk/manim.x86_64-linux

This is indeed kind of puzzling. I just tried to build v0.16.0 and v.16.0.post0 on my system from up-todate nixpkgs-master with the same result:

```
=========================== short test summary info ============================
FAILED tests/test_graphical_units/test_mobjects.py::test_PointCloudDot
FAILED tests/test_graphical_units/test_threed.py::test_Torus
== 2 failed, 541 passed, 1 skipped, 5 xfailed, 1 xpassed in 289.21s (0:04:49) ==
Manim Community v0.16.0

error: builder for '/nix/store/lyy58z5knih7b5rkcn2j28pbl66s8h3v-manim-0.16.0test.drv' failed with exit code 1;
       last 10 log lines:
       > E           Max absolute difference: 255
       > E           Max relative difference: 2.42857143
       >
       > manim/utils/testing/_frames_testers.py:42: AssertionError
       > =========================== short test summary info ============================
       > FAILED tests/test_graphical_units/test_mobjects.py::test_PointCloudDot
       > FAILED tests/test_graphical_units/test_threed.py::test_Torus
       > == 2 failed, 541 passed, 1 skipped, 5 xfailed, 1 xpassed in 289.21s (0:04:49) ==
       > Manim Community v0.16.0
       >
       For full logs, run 'nix log /nix/store/lyy58z5knih7b5rkcn2j28pbl66s8h3v-manim-0.16.0test.drv'.
```

**Comment by friedelino** [comment]
I just had a look at [**Hydra's**](https://hydra.nixos.org/build/187896077/log) last build log test session section:
```
============================= test session starts ==============================
platform linux -- Python 3.10.6, pytest-7.1.2, pluggy-1.0.0
rootdir: /build/source, configfile: pyproject.toml
plugins: xdist-2.5.0, forked-1.4.0, anyio-3.5.0
gw0 [550] / gw1 [550]m
........................................................................ [ 13%]
........................................................................ [ 26%]
........................................................................ [ 39%]
............s.................................................xxxx.x.... [ 52%]
........................................................................ [ 65%]
........................................................................ [ 78%]
........................................................................ [ 91%]
.......................X......................                           [100%]
======= 543 passed, 1 skipped, 5 xfailed, 1 xpassed in 490.37s (0:08:10) =======
```

... and i was wondering why it looks different to **my** build log :thinking: :

```
============================= test session starts ==============================
platform linux -- Python 3.10.6, pytest-7.1.2, pluggy-1.0.0
rootdir: /build/source, configfile: pyproject.toml
plugins: xdist-2.5.0, forked-1.4.0, anyio-3.5.0
gw0 [550] / gw1 [550] / gw2 [550] / gw3 [550] / gw4 [550] / gw5 [550] / gw6 [550] / gw7 [550]m
........................................................................ [ 13%]
........................................................................ [ 26%]
......................................................................s. [ 39%]
......................................x................................. [ 52%]
...............x..x...x..x.........................................F.... [ 65%]
........................................................................ [ 78%]
.....................................................................X.. [ 91%]
......................................F.......                           [100%]
=================================== FAILURES ===================================
______________________________ test_PointCloudDot ______________________________
[gw5] linux -- Python 3.10.6 /nix/store/xf1k5k05vg3zn7dfcpfh1qa7ga48hi3m-python3-3.10.6/bin/python3.10
...
__________________________________ test_Torus __________________________________
[gw6] linux -- Python 3.10.6 /nix/store/xf1k5k05vg3zn7dfcpfh1qa7ga48hi3m-python3-3.10.6/bin/python3.10
...
=========================== short test summary info ============================
FAILED tests/test_graphical_units/test_mobjects.py::test_PointCloudDot
FAILED tests/test_graphical_units/test_threed.py::test_Torus
== 2 failed, 541 passed, 1 skipped, 5 xfailed, 1 xpassed in 289.21s (0:04:49) ==
```
Should the tests not have the same results?

My complete build log: [manim.log](https://github.com/NixOS/nixpkgs/files/9387198/manim.log)

**Comment by friedelino** [comment]
:bulb: Aha!

I think i found the answer... and maybe also an reproducibility-issue when using `pytest-xdist` with nix.
[pytest-xdist](https://pytest-xdist.readthedocs.io/en/latest/index.html) distributes test execution randomly over multiple CPU cores when invoked with the `-n auto` flag. In my case this leads to running tests on 8 cores, in Hydra's case on only 2. As i found out in the issues-section of pytest-xdist there are some issues with tests involving randomization, because test results of different cores might be compared, which sometimes leads to failures.

**Comment by smancill** [comment]
Nice debugging.

Then this should be OK to be merged.

**Review by smancill — APPROVED** [review]

**Pull request merged**